code/modules/atmos_automation/implementation/vent_pump.dm

```diff
@@ -88,4 +88,202 @@
+				filter = RADIO_FROM_AIRALARM
+			parent.send_signal(data, filter)
+
+	GetText()
+		if(mode)//DP vent
+			return {"Set <a href=\"?src=\ref[src];swap_modes=1\">dual-port</a> vent pump <a href=\"?src=\ref[src];set_vent_pump=1\">[fmtString(vent_pump)]</a>
+				pressure bounds: internal outwards: <a href=\"?src=\ref[src];set_intpressure_out=1">[fmtString(intpressureout)]</a>
+				internal inwards: <a href=\"?src=\ref[src];set_intpressure_in=1">[fmtString(intpressurein)]</a>
+				external: <a href=\"?src=\ref[src];set_external=1">[fmtString(extpressure)]</a>
+			"}//well that was a lot to type
+		else
+			return {"Set <a href=\"?src=\ref[src];swap_modes=1\">unary</a> vent pump <a href=\"?src=\ref[src];set_vent_pump=1\">[fmtString(vent_pump)]</a>
+				pressure bounds: internal: <a href=\"?src=\ref[src];set_intpressure_out=1">[fmtString(intpressureout)]</a>
+				external: <a href=\"?src=\ref[src];set_external=1">[fmtString(extpressure)]</a>
+			"}//copy paste FTW
+
+	Topic(href, href_list)
+		if(..())
+			return 1
+
+		if(href_list["set_vent_pump"])
+			var/list/injector_names=list()
+			if(mode)//DP vent selection
+				for(var/obj/machinery/atmospherics/binary/dp_vent_pump/I in world)
+					if(!isnull(I.id_tag) && I.frequency == parent.frequency)
+						injector_names|=I.id_tag
+			else
+				for(var/obj/machinery/atmospherics/unary/vent_pump/I in machines)
+					if(!isnull(I.id_tag) && I.frequency == parent.frequency)
+						injector_names|=I.id_tag
+			vent_pump = input("Select a vent:", "Vent Pumps", vent_pump) as null|anything in injector_names
+			parent.updateUsrDialog()
+			return 1
+
+		if(href_list["set_intpressure_out"])
+			var/response = input("Set new pressure, in kPa. \[0-[50*ONE_ATMOSPHERE]\]") as num
+			intpressureout = text2num(response)
+			intpressureout = Clamp(intpressureout, 0, 50*ONE_ATMOSPHERE)
+			parent.updateUsrDialog()
+			return 1
+
+		if(href_list["set_intpressure_in"])
+			var/response = input("Set new pressure, in kPa. \[0-[50*ONE_ATMOSPHERE]\]") as num
+			intpressurein = text2num(response)
+			intpressurein = Clamp(intpressurein, 0, 50*ONE_ATMOSPHERE)
+			parent.updateUsrDialog()
+			return 1
+
+		if(href_list["set_external"])
+			var/response = input(usr,"Set new pressure, in kPa. \[0-[50*ONE_ATMOSPHERE]\]") as num
+			extpressure = text2num(response)
+			extpressure = Clamp(extpressure, 0, 50*ONE_ATMOSPHERE)
+			parent.updateUsrDialog()
+			return 1
+
+		if(href_list["swap_modes"])
+			mode = !mode
+			vent_pump = null//if we don't clear this is could get glitchy, by which I mean not at all, whatever, stay clean
+			parent.updateUsrDialog()
+			return 1
+
+/datum/automation/set_vent_pressure_checks
+	name = "Vent pump: pressure checks"
+
+	var/vent_pump = null
+	var/checks = 1
+	var/mode = 0//1 for DP vent, 0 for unary vent
+/*
+checks bitflags
+1 = external
+2 = internal in (regular internal for unaries)
+4 = internal out (ignored by unaries)
+*/
+
+
+	Export()
```

Comment:
The pathing in all the other automation defines was relative too, I just followed it like it was